### PR TITLE
chore: added the engines property to the package.json

### PR DESCRIPTION
**Pull request by viraj-s15**
Signed-off-by: viraj-s15 <veer15102003@gmail.com>

## Proposed Changes
Added the engines property to the package.json
## Description
Fixes: https://github.com/cloudevents/sdk-javascript/issues/527

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,5 +155,8 @@
   "publishConfig": {
     "access": "public"
   },
-  "types": "./dist/index.d.ts"
+  "types": "./dist/index.d.ts",
+  "engines": {
+    "node": ">=14.0.0 <20.0.0"
```
Comment (lance) [inline package.json:160]:
```suggestion
    "node": ">=12 <20.0.0"
```
We actually support v12.x and up

Comment (viraj-s15) [reply package.json]: Ahh I apologise. Reading the previous comment got me a little confused.

File: package.json [hunk 2]
```diff
@@ -155,5 +155,8 @@
   "publishConfig": {
     "access": "public"
   },
-  "types": "./dist/index.d.ts"
+  "types": "./dist/index.d.ts",
+  "engines": {
+    "node": ">=12 <20.0.0"
```
Comment (lholmquist) [inline package.json:160]: Node 12 is EOL'd, but for us to drop it, that would require a major version bump,  which we can do in a separate PR,  so this is good.  

Not really a blocker,  but my preference would be to only have ">=12" and not the "<20.0.0"

**Review by lholmquist — APPROVED** [review]
Just need to sign the commit,  but LGTM

**Comment by lholmquist** [comment]
closed via https://github.com/cloudevents/sdk-javascript/commit/bc3aaca2ef250e4acd72b909488b326233237c83
